### PR TITLE
Added separate mintToPool and mintToWriter functions for options (and correspondingly to obligations)

### PR DESCRIPTION
**Pull request by theodorebugnet**
Additionally, added lockAndWriteToSelf and unlockAndMintToWriter to the lib to make use of the new functions.

### CODE REVIEW

File: contracts/OptionLib.sol [hunk 1]
```diff
@@ -94,7 +94,22 @@ contract OptionLib is UniswapV2Router02 {
...
-        IObligation(obligation).mint(msg.sender, optionsAmount);
+        IObligation(obligation).mintToPool(msg.sender, optionsAmount);
+    }
+
+    /// @notice Atomically deposit collateral into a treasury and directly credit
+    /// the writer with the resulting options.
+    /// @param obligation The obligation tokens
+    /// @param collateral The collateral tokens
+    /// @param optionsAmount The amount of options that will be minted (hence the amount of collateral that must be locked)
+    function lockAndWriteToSelf(
```
Comment (gregdhill) [inline contracts/OptionLib.sol]:
```suggestion
    function lockAndWriteToWriter(
```

Comment (gregdhill) [reply contracts/OptionLib.sol]: Let's equate the naming conventions, can you also rename the one above to `lockAndWriteToPool`?

**Review by gregdhill — COMMENTED** [review]
Minor naming nit pick, otherwise LGTM!

File: contracts/OptionLib.sol (hunk 1 above)
Comment (theodorebugnet) [reply contracts/OptionLib.sol]: My reasoning was that it credits msg.sender as the writer, but you definitely have a point with keeping to convention, updating

**Review by gregdhill — APPROVED** [review]

**Pull request merged**